File: internal/model/app_settings.go
package model

// AutoLabelRule is a single auto-label match rule: when a task title contains
// Mask, the labels with the listed IDs are attached. If IgnoreCase is true,
// comparison is case-insensitive. Labels are never auto-created — IDs must
// reference existing labels; missing IDs are silently skipped at apply time.
type AutoLabelRule struct {
	Mask       string  `json:"mask"`
	LabelIDs   []int64 `json:"labelIds"`
	IgnoreCase bool    `json:"ignoreCase"`
}

// ProjectSuggestionRule is a single project-suggestion match rule: when a task
// title contains Mask, the projects with the listed IDs are offered to the user
// as a choice. If IgnoreCase is true, comparison is case-insensitive. Unlike
// auto-labels, nothing is applied automatically — the rules only feed the
// suggestion chips in the quick-add dialog; missing IDs are silently skipped.
type ProjectSuggestionRule struct {
	Mask       string  `json:"mask"`
	ProjectIDs []int64 `json:"projectIds"`
	IgnoreCase bool    `json:"ignoreCase"`
}

// AppSettings holds global, server-wide settings persisted in the app_settings
// table (single-row, id=1).
type AppSettings struct {
	AutoLabels         []AutoLabelRule         `json:"autoLabels"`
	ProjectSuggestions []ProjectSuggestionRule `json:"projectSuggestions"`
}

File: android-native/core/model/src/main/kotlin/ru/tinyops/turboist/core/model/AppSettings.kt
package ru.tinyops.turboist.core.model

/**
 * A rule that attaches labels to a task whose title contains [mask]. Labels are
 * never created by a rule: an id that no longer resolves is skipped.
 *
 * @property labelIds **server** label ids — the rules blob is written by the
 *   server and echoed back unchanged.
 */
data class AutoLabelRule(
    val mask: String,
    val labelIds: List<Long> = emptyList(),
    val ignoreCase: Boolean = false,
)

/**
 * A rule that offers projects for a task whose title contains [mask]. Unlike an
 * auto-label rule nothing is applied: the matches only become suggestions the
 * user can accept while typing.
 *
 * @property projectIds **server** project ids, for the same reason as
 *   [AutoLabelRule.labelIds].
 */
data class ProjectSuggestionRule(
    val mask: String,
    val projectIds: List<Long> = emptyList(),
    val ignoreCase: Boolean = false,
)

/**
 * Server-wide settings, one row for the whole installation — distinct from
 * [UserSettings], which is the user's own preferences. The two are never merged:
 * they have different owners and different write paths.
 */
data class AppSettings(
    val autoLabels: List<AutoLabelRule> = emptyList(),
    val projectSuggestions: List<ProjectSuggestionRule> = emptyList(),
)
